notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/filter/ReadStateSupportFilter.java

```diff
@@ -120,8 +123,24 @@ public NotificationResponse doFilter(HttpServletRequest request, INotificationSe
             }
         }
 
-        return rslt;
+        if (StringUtils.isNotBlank(readFilterParameter)) {
+            boolean readFilterValue = Boolean.parseBoolean(readFilterParameter);
+
+            return rslt.filter(
+                    entry -> {
+                        boolean isRead =
+                                entry.getAttributes()
+                                        .stream()
+                                        .anyMatch(
+                                                event ->
```

Comment:
This change looks 95% great, but I think there's a mismatch here.

IIRC the object passed to `anyMatch()` will be of type `NotificationAttribute`, and as such it is tremendously unlikely to produce a `true` response when passed to `NotificationState.READ.equals()`.

Reply:
I addressed the mismatch. Thanks for catching it!